### DISCUSSION TITLE
Whitelist TPR Android

### DISCUSSION THREAD

**Discussion by Devamitta** [bksubhuti/tipitaka-pali-reader#233] [Q&A]
Bhante, 

can you please recommend which HTTPS URLs we need to whitelist so we can download updates for the TPR Android extensions? Where does it download updates from? For example latest DPD or Pali Englisg 4 Sutta Nikaya .

**Comment by bksubhuti** [comment]
whitelist https://github.com/bksubhuti/

**Comment by bksubhuti** [comment]
external links also exist to americanmonk.org for help
youtube for the channel
android playstore apple store and microsoft app store and flathub.org

**Comment by Devamitta** [comment]
Thank you Bhante.

And how to use extension with English translation on android? How I can open the translation for some sutta?

**Comment by bksubhuti** [comment]
You can watch the videos..
It is on the menu for youtube.. enjoy.!
 Please close this issue.

**Comment by Devamitta** [comment]
Bhante, related to this:

We have successfully installed TPR on the Windows PC, and it is functioning properly. However, an internet connection is required to download extensions. We have here limited whitelisted websites for indriyasaṃvara purposes. Github has already been whitelisted. What else do we need to whitelist in order to be able to download extensions?

**Comment by bksubhuti** [comment]
I'm sorry.. not sure how to help you.. If you can get permission to temporarily use a vpn to download the extensions.  Perhaps that can help you.  There is nothing I can do about internet problems.  One thought.  The software downloads zip files.  It sounds like you have strong network restrictions.  It could be that your network admin blocks zip files which can strain the system with large downloads.  You can ask whoever is in charge or to temporarily turn that restriction off from time to time (if that is the case).
You will need to do this each time you upgrade with a new DB.  The next release will be such a case.

**Comment by Devamitta** [comment]
Thank you Bhante for your attention.
The main problem here is that the software downloads those extensions for every user individually. With many residents who may use TPR it can be quite difficult.
If would be possible to know HTTP address from which the software downloads those zip files, it would be already good enough to put those HTTP in the whitelist.
